CustomEstimator/modules/ensemble_modules/trainer_from_storage/trainer/task.py

```diff
@@ -1,173 +1 @@
                                 random_state=19,
                                 is_trial=is_trial)

    ##
    model.create_ensemble_architecture(hidden_units=hidden_units,
                                       n_output=y_train.shape[1],
                                       primary_models_directory=primary_models_directory,
                                       images_shape=images_shape,
                                       save_path=ensemble_architecture_path)

    print('The ensemble architecture was made and is ready to be used.')

    classifier = tf.estimator.Estimator(
        model_fn=model.model_fn,
        params={
            'primary_models_directory': primary_models_directory,
            'images_shape': images_shape,
            'hidden_units': hidden_units,
            'learning_rate': learning_rate,
            'ensemble_architecture_path': ensemble_architecture_path,
            'retrain_primary_models': retrain_primary_models,
            'category_map': map
        },
        model_dir=job_dir)

    # Train and evaluate model.

    ##
    classifier.train(input_fn=lambda: Dataset.prep_input_function(
        prefetch_buffer_size=prefetch_buffer_size,
        train_epochs=train_epochs,
        multi_threading=image_processing_multi_threading,
        train_batch_size=batch_size,
        mode=tf.estimator.ModeKeys.TRAIN,
        X_train_path_names=X_train_path_names,
        X_test_path_names=X_test_path_names,
        y_train=y_train,
        y_test=y_test
    ))

    eval_result = classifier.evaluate(input_fn=lambda: Dataset.prep_input_function(
        multi_threading=image_processing_multi_threading,
        mode=tf.estimator.ModeKeys.EVAL,
        X_train_path_names=X_train_path_names,
        X_test_path_names=X_test_path_names,
        y_train=y_train,
        y_test=y_test))

    def serving_input_receiver_func():
        images_str = tf.placeholder(tf.string, shape=[None], name='export_input_image_bytes')
        def decode_and_resize(image_str_tensor):
            image = tf.image.decode_jpeg(image_str_tensor, channels=3)
            image = tf.expand_dims(image, 0)
            image = tf.image.resize_bilinear(image, [224, 224], align_corners=False)
            image = tf.squeeze(image, squeeze_dims=[0])
            return image
        images = tf.map_fn(decode_and_resize, images_str, dtype=tf.float32)
        return tf.estimator.export.ServingInputReceiver({'img': images}, {'bytes': images_str})

    classifier.export_savedmodel(export_dir_base=export_dir,
                                 serving_input_receiver_fn=serving_input_receiver_func)

    print('')
    print('')
    print('validation accuracy after {} epochs is: {}'.format(train_epochs, eval_result['accuracy']))
    print('')
    print('')


##
args_parser = argparse.ArgumentParser()
HYPER_PARAMS = initialise_hyper_params(args_parser)

##
if __name__ == '__main__':
    tf.logging.set_verbosity(tf.logging.INFO)
    tf.app.run(main)

```

Comment:
Can you review and format this file? It appears to all be on one lime.

Reply:
Changes applied.